### ISSUE TITLE
Plurals fails to compile with duplicate `__var_count` generic

### ISSUE THREAD

**Issue by lukexor** [Baptistemontan/leptos_i18n#84]
Been struggling to get this to compile in my project. I'm not sure if I'm just using this feature wrong, or if there's a bug in the implementation.

It seems to be caused when there is only one locale available.

Here's a minimal reproduction:

`Cargo.toml`
```toml
[dependencies]
leptos = { version = "0.5.0", features = ["csr"] }
leptos_meta = { version = "0.5.0", features = ["csr"] }
leptos_i18n = { path = "../../leptos_i18n", default-features = false, features = [
    "debug_interpolations",
    "csr",
    "cookie",
    "yaml_files",
] }
serde = { version = "1", features = ["derive"] }
console_error_panic_hook = { version = "0.1" }
wasm-bindgen = { version = "=0.2.87" }

[package.metadata.leptos-i18n]
default = "en"
locales = ["en"]
```

`src/app.rs`
```rust
#[component]
pub fn App() -> impl IntoView {
    leptos_meta::provide_meta_context();

    let i18n = provide_i18n_context();

    view! {
        <p>{t!(i18n, result_summary, count = || 5, query_str = || "test")}</p>
    }
}
```

`locales/en.yaml`
```yaml
---
result_summary:
  - "u64"
  - ["No results for {{ query_str }}:", 0]
  - ["1 result for {{ query_str }}:", 1]
  - ["{{ count }} results for {{ query_str }}:"]
```

```txt
❯ trunk build
2023-10-27T22:24:07.047974Z  INFO 📦 starting build
2023-10-27T22:24:07.048390Z  INFO spawning asset pipelines
2023-10-27T22:24:07.160652Z  INFO building yaml
   Compiling yaml v0.1.0 (/Users/lpetherbridge/pdev/leptos_i18n/examples/yaml)
error[E0403]: the name `__var_count` is already used for a generic parameter in this item's generic parameters
 --> src/main.rs:4:1
  |
4 | leptos_i18n::load_locales!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | already used
  | first use of `__var_count`
  |
  = note: this error originates in the macro `leptos_i18n::load_locales` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0416]: identifier `var_count` is bound more than once in the same pattern
 --> src/main.rs:4:1
  |
4 | leptos_i18n::load_locales!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ used in a pattern more than once

error[E0124]: field `var_count` is already declared
 --> src/main.rs:4:1
  |
4 | leptos_i18n::load_locales!();
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  | |
  | field already declared
  | `var_count` first declared here
  |
  = note: this error originates in the macro `leptos_i18n::load_locales` (in Nightly builds, run with -Z macro-backtrace for more info)

Some errors have detailed explanations: E0124, E0403, E0416.
For more information about an error, try `rustc --explain E0124`.
error: could not compile `yaml` (bin "yaml") due to 3 previous errors
```

**Comment by Baptistemontan** [comment]
Yup plurals are indeed brokens, thanks for the issue, I'll fix work on a fix right now

**Comment by Baptistemontan** [comment]
Ok I get where this is coming from, the removal of conflicting variable vs plural count is done when merging locales keys, but in your exemple there is only 1 locale, so no merging is done...

**Comment by Baptistemontan** [comment]
just released `v0.2.3` with the fix

**Comment by lukexor** [comment]
So fast! Thanks!
